### PR TITLE
Exosphere publish script

### PR DESCRIPTION
**Pull request by hugoesthere**
@kevgo @trushton

### CODE REVIEW

File: bin/publish [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                            exo-clone
+                            exo-create
+                            exo-deploy
+                            exo-lint
+                            exo-run
+                            exo-setup
+                            exo-sync
+                            exo-test
+                            exosphere-shared
+                            website          ]>
```
Comment (hugoesthere) [inline bin/publish:22]: Tests for `website` still do not pass at this point, so technically this publish script won't finish.

File: bin/publish [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  if process.argv.length != 3
+    display-help!
+
+  @level = process.argv[2]
+  display-help! if @level in ['-h, --help']
+
+  @target-version = get-target-version version, @level
+
+  <- confirm-target-version
+  # ensure-no-open-files!
```
Comment (kevgo) [inline bin/publish]: Why is this stuff commented out?

File: bin/publish [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+      replace:
+        /"exorelay": "\d+.\d+.\d+"/g
+        /"exoservice": "\d+.\d+.\d+"/g
+        /"exocom-mock": "\d+.\d+.\d+"/g
+        /"exocom-dev": "\d+.\d+.\d+"/g
+      with:
+        "\"exorelay\": \"#{@target-version}\""
+        "\"exoservice\": \"#{@target-version}\""
+        "\"exocom-mock\": \"#{@target-version}\""
+        "\"exocom-dev\": \"#{@target-version}\""
```
Comment (kevgo) [inline bin/publish:78]: This is such cool functionality!

**Review by kevgo — APPROVED** [review]

File: bin/publish (hunk 2 above)
Comment (hugoesthere) [reply bin/publish]: Woops, leftover from developing on my machine.

**Comment by hugoesthere** [comment]
NOTE: this PR also contains version bumps because I was testing the publish script on this branch (before shipping out to master). In the future, the publish script can be run only on master and those version bumps will be committed to master.

**Pull request closed without merging**